code/modules/mob/living/say.dm

```diff
@@ -231,7 +231,7 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 /mob/living/IsVocal()
 	. = ..()
 
-	if(disabilities & MUTE)
+	if(HAS_TRAIT(src, TRAIT_MUTED) || disabilities & MUTE)
 		return FALSE
```

Comment:
Use eiher the disabilities var or traits not both

For example you can make the bitflag be toggled by the status effect